Add configurable sleep timings.

sw1nn opened the pull request:
* Allow a fn to be specified to generate sleep values between tries.
* Provide a fn to generate a exponential backoff with jitter strategy
  sleep fn.

sw1nn committed:
Add configurable sleep timings.

* Allow a fn to be specified to generate sleep values between tries.
* Provide a fn to generate a exponential backoff with jitter strategy
  sleep fn.

arohner commented on test/circle/wait_for8_test.clj:
```diff
@@ -239,3 +239,28 @@
       (is (= 1 (-> foo bond/calls count))))))
+
+(deftest backoff-fn
+  (testing "never-negative"
```
If we care about never negative, should the spec include `pos?`

sw1nn committed:
Improve conformance testing of exponential-backoff-fn

We define a spec for the fn that can be passed as a :sleep arg and
also rework the test of exponential-backoff-fn to test explicitly
against that spec.

sw1nn committed:
Change project id and release repository.

sw1nn committed:
Version 1.0.6

sw1nn committed:
Version 1.0.7-SNAPSHOT
